### ISSUE TITLE
PathPattern vs IdentifierPattern

### ISSUE THREAD

**Issue by MemPage** [rust-lang-nursery/reference#605]
How to differ between _IdentifierPattern_ and _PathPattern_ when they can be just _IDENTIFIER_?

Rules applied:
```
Pattern: IdentifierPattern
Pattern->IdentifierPattern

IdentifierPattern: ref? mut? IDENTIFIER (@ Pattern ) ?
IdentifierPattern->IDENTIFIER
```

And for the _PathPattern_
```
PathPattern: PathInExpression
PathPattern->PathInExpression

PathInExpression: ::? PathExprSegment (:: PathExprSegment)*
PathInExpression->PathExprSegment

PathExprSegment:PathIdentSegment (:: GenericArgs)?
PathExprSegment->PathIdentSegment

PathIdentSegment:IDENTIFIER
PathIdentSegment->IDENTIFIER
```
Not sure if there is a BIGGER rule to solve this out, or just we don't care at the parsing stage, but how to decide is this _IDENTIFIER_ a _IdentifierPattern_ or _PathPattern_?

**Comment by ehuss** [comment]
I don't think you can disambiguate with pure parsing (see [PatKind](https://github.com/rust-lang/rust/blob/a3404557c54ea48fb8efc805d93c450beb3364d4/src/libsyntax/ast.rs#L627-L630)), it has to be done with name resolution.  I *think* this particular disambiguation is done [here](https://github.com/rust-lang/rust/blob/a3404557c54ea48fb8efc805d93c450beb3364d4/src/librustc/hir/lowering.rs#L3945-L3970), though I am not very familiar with `rustc`.
